coral-component-select/src/scripts/Select.js

```diff
@@ -482,24 +484,52 @@ class Select extends BaseFormField(BaseComponent(HTMLElement)) {
     this._elements.taglist.readOnly = this._readOnly || isDisabled;
     this._elements.taglist.disabled = this._readOnly || isDisabled;
   }
+
+  /**
+   Inherited from {@link BaseFormField#labelled}.
+   */
+  get labelled() {
+    return super.labelled;
+  }
+  set labelled(value) {
+    super.labelled = value;
+  
+    if (this.labelled) {
+      if (!this.labelledBy) {
+        this._elements.button.setAttribute('aria-labelledby', `${this._elements.button.id} ${this._elements.label.id} ${this.invalid ? this._elements.invalidIcon.id : ''}`);
```

Comment:
What is the point of setting the `aria-label` of button to the id of button?

Reply:
If someone sets label using `labelled` prop, `aria-label` gets added to the `button` element, which will override any of the descendant content. This way, we can label the control using the `button[id][aria-label]`, its `label`, and if necessary, the `invalid` icon.